pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -0,0 +1,3257 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from math import log10
+from enum import Enum, IntFlag
+from numpy import arange
+from datetime import datetime
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, truncated_discrete_set, \
+    joined_validators, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class StrEnum(str, Enum):
```

Comment:
Wait, this is overriding the `StrEnum` name from `enum.StrEnum` (if you are on py 3.11+), do we really want that?
If yes, I'd be more comfortable with a version switch like 
```
if sys.version_info >= (3, 11):
    from enum import StrEnum
else:
    your solution
```
plus a comment to remove it when our minimum version becomes 3.11.
but ideally we'd use packages for that. I have found but not investigated, https://pypi.org/project/backports.strenum/

Reply:
another idea:
~~~python
try:
    from enum import StrEnum
except ImportError:
   # define it yourself
~~~

Reply:
Yeah, that achieves the same thing as the upper suggestion. I like the first one better because it's more explicit -- it's clear _why_ we are checking and choosing one or the other.